rmw_zenoh_cpp/src/rmw_zenoh.cpp

```diff
@@ -3063,11 +2757,8 @@ check_and_attach_condition(
 
   if (subscriptions) {
     for (size_t i = 0; i < subscriptions->subscriber_count; ++i) {
-      auto sub_data =
-        static_cast<rmw_zenoh_cpp::rmw_subscription_data_t *>(subscriptions->subscribers[i]);
-      if (sub_data == nullptr) {
-        continue;
-      }
+      rmw_zenoh_cpp::SubscriptionData * sub_data =
+        static_cast<rmw_zenoh_cpp::SubscriptionData *>(subscriptions->subscribers[i]);
```

Comment:
I think we still want to check whether `sub_data` is NULL before doing anything:

```suggestion
      rmw_zenoh_cpp::SubscriptionData * sub_data =
        static_cast<rmw_zenoh_cpp::SubscriptionData *>(subscriptions->subscribers[i]);
      if (sub_data == nullptr) {
        continue;
      }
```